src/App.tsx

```diff
@@ -1,24 +1,149 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo, FilterType } from './types/Todo';
+import { AddInput } from './components/AddInput/AddInput';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Loading } from './components/isLoading';
+import { TodoList } from './components/TodoList';
+import { AddError } from './components/AddError/AddError';
+import {
+  getTodos,
+  deleteTodos,
+  postTodos,
+} from './api/todos';
+import { TodoItem } from './components/TodoItem';
+import { createTitle } from './utils/helpers';
 
-const USER_ID = 0;
+const USER_ID = 7025;
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disInput, setDisInput] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const fetchData = async () => {
+    const todosFromServer = await getTodos(USER_ID);
+
+    try {
+      setTodo(todosFromServer);
+      setIsLoading(false);
+    } catch {
+      setError('Unable to load a todo');
+      setIsLoading(true);
+    }
+  };
+
+  const filterTodo = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterType) {
+        case FilterType.ALL:
+          return todo;
+
+        case FilterType.ACTIVE:
+          return !todo.completed;
+
+        case FilterType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterType]);
+
+  const handleAddTodo = async (title: string) => {
+    try {
+      setDisInput(true);
+      setTempTodo(createTitle(title));
+
+      const data = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      const newTodo = await postTodos(data);
+
+      setTodo(todo => [...todo, newTodo]);
+    } catch {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setDisInput(false);
+    }
+  };
+
+  const handleDeleteTodo = async (id: number) => {
+    try {
+      await deleteTodos(id);
+      const currentTodo = todos.filter((todo) => todo.id !== id);
+
+      setTodo(currentTodo);
+      setError('');
+    } catch {
+      setError('Unable to delete a todo');
+    }
+  };
+
+  useEffect(() => {
+    if (error) {
```

Comment:
The useEffect should have 'error' as a dependency to avoid unnecessary re-renders.
And don't forget clear timeout in unmount effect

Reply:
where is unmount?

Reply:
not fixed